server/http.go

```diff
@@ -53,11 +59,152 @@ func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Req
 		p.completeUserOAuthToZoom(w, r)
 	case "/deauthorization":
 		p.deauthorizeUser(w, r)
+	case "/api/v1/updatePMI":
+		p.setPMI(w, r)
+	case "/api/v1/askPMI":
+		p.askPMI(w, r)
 	default:
 		http.NotFound(w, r)
 	}
 }
 
+func (p *Plugin) GetPostWithSlackAttachment(mattermostUserID string, attachments ...*model.SlackAttachment) (*model.Post, error) {
+	channel, err := p.API.GetDirectChannel(mattermostUserID, p.botUserID)
+	if err != nil {
+		p.API.LogInfo("Couldn't get bot's DM channel", "user_id", mattermostUserID)
+		return nil, err
+	}
+
+	post := &model.Post{
+		ChannelId: channel.Id,
+		UserId:    p.botUserID,
+	}
+
+	model.ParseSlackAttachment(post, attachments)
+
+	return post, nil
+}
+
+func (p *Plugin) askPMI(w http.ResponseWriter, r *http.Request) {
+	response := &model.PostActionIntegrationResponse{}
+	decoder := json.NewDecoder(r.Body)
+	postActionIntegrationRequest := &model.PostActionIntegrationRequest{}
+	if err := decoder.Decode(&postActionIntegrationRequest); err != nil {
+		p.API.LogError("Error decoding PostActionIntegrationRequest params.", "Error", err.Error())
+		w.Header().Set("Content-Type", "application/json")
+		if err := json.NewEncoder(w).Encode(response); err != nil {
+			p.API.LogWarn("failed to write response", "error", err.Error())
+		}
+		return
+	}
+
+	action := postActionIntegrationRequest.Context["action"].(string)
+	userID := postActionIntegrationRequest.Context["userID"].(string)
+	channelID := postActionIntegrationRequest.Context["channelID"].(string)
+
+	slackAttachment := model.SlackAttachment{
+		Text: fmt.Sprintf("You have selected `%s` to start a meeting.", action),
+	}
+
+	post := &model.Post{
+		ChannelId: channelID,
+		UserId:    p.botUserID,
+		Id:        postActionIntegrationRequest.PostId,
+		CreateAt:  time.Now().Unix() * 1000,
+	}
+
+	model.ParseSlackAttachment(post, []*model.SlackAttachment{&slackAttachment})
+	p.API.UpdateEphemeralPost(userID, post)
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(post); err != nil {
+		p.API.LogWarn("failed to write response", "error", err.Error())
+	}
+
+	p.startMeeting(action, userID, channelID)
+}
+
+func (p *Plugin) startMeeting(action string, userID string, channelID string) {
+	user, appErr := p.API.GetUser(userID)
+	if appErr != nil {
+		return
```

Comment:
we are not returning these errors atleast log them so in case of error user will be able to find the error